src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,45 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    private final LocalDate byDate;
+    private final String byText;
+
+    private static final DateTimeFormatter DISPLAY_FMT = DateTimeFormatter.ofPattern("MMM d yyyy");
+
+    public Deadline(String description, LocalDate byDate) {
+        super(description);
+        if (byDate == null) {
+            throw new DukeException("OOPS!!! Deadline date cannot be null.");
+        }
+        this.byDate = byDate;
+        this.byText = byDate.toString();
+    }
+
+    public Deadline(String description, String byText) {
+        super(description);
+        if (byText == null || byText.isBlank()) {
+            throw new DukeException("OOPS!!! Deadline must have a /by value.");
+        }
+        this.byText = byText.trim();
+        LocalDate parsed = null;
```

Comment:
Perhaps the name of this variable could be improved to capture what is being parsed?
```suggestion
        LocalDate parsed = null;
```